v3/src/data-interactive/handlers/adornment-handler.ts

```diff
@@ -1,32 +1,46 @@
-import { getSnapshot } from "@concord-consortium/mobx-state-tree"
+import { IAdornmentModel } from "../../components/graph/adornments/adornment-models"
+import { IGraphContentModel, isGraphContentModel } from "../../components/graph/models/graph-content-model"
 import { registerDIHandler } from "../data-interactive-handler"
 import { DIHandler, DIResources } from "../data-interactive-types"
 import { adornmentNotFoundResult } from "./di-results"
-import { t } from "../../utilities/translation/translate"
 
+export interface DIAdornmentHandler {
+  get: (adornment: IAdornmentModel, component: IGraphContentModel) => Maybe<Record<string, any>>
+}
+
+const diAdornmentHandlers = new Map<string, DIAdornmentHandler>()
+
+const diAdornmentTypeAliases = new Map<string, string>()
+
+export const registerAdornmentHandler = (type: string, handler: DIAdornmentHandler, alias?: string) => {
+  diAdornmentHandlers.set(type, handler)
+  if (alias) diAdornmentTypeAliases.set(alias, type)
+  const trimType = type.replace(/\s/g, "")
+  // register trimmed (without spaces) types as aliases
+  if (trimType !== type) diAdornmentTypeAliases.set(trimType, type)
+}
+
+export const resolveAdornmentType = (typeOrAlias: unknown) => {
+  return typeof typeOrAlias === "string"
+          ? diAdornmentTypeAliases.get(typeOrAlias) ?? typeOrAlias
+          : typeOrAlias
+}
 
 export const diAdornmentHandler: DIHandler = {
   get(resources: DIResources) {
-    const { adornment } = resources
-    if (!adornment) return adornmentNotFoundResult
-
-    const adornmentSnapshot = getSnapshot(adornment)
-
-    // Translate `labelTitle` if it exists.
-    if ("labelTitle" in adornmentSnapshot && typeof adornmentSnapshot.labelTitle === "string") {
-      return {
-        success: true,
-        values: {
-          ...adornmentSnapshot,
-          labelTitle: t(adornmentSnapshot.labelTitle)
-        }
-      }
+    const { adornment, component } = resources
+    if (!adornment?.isVisible) return adornmentNotFoundResult
```

Comment:
Is this going to confuse plugin developers? It seems weird that these would be included in a `get adornmentList` request, but they can't be found in a `get adornment` request.